<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SongInfo</class>
 <widget class="QWidget" name="SongInfo">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>672</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="windowOpacity">
   <double>0.500000000000000</double>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <item>
      <widget class="QWidget" name="songChanged" native="true">
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: yellow</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0,0,0,0">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;The song has been modified!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="revertSong">
          <property name="text">
           <string>Revert</string>
          </property>
          <property name="icon">
           <iconset theme="document-revert">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="undoButton_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="toolTip">
           <string>Undo</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset theme="edit-undo">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="redoButton_2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Redo</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset theme="edit-redo">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="saveSong">
          <property name="text">
           <string>Save</string>
          </property>
          <property name="icon">
           <iconset theme="document-save">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::East</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Triangular</enum>
       </property>
       <property name="currentIndex">
        <number>2</number>
       </property>
       <widget class="QWidget" name="generalTab">
        <attribute name="title">
         <string>General</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QScrollArea" name="scrollArea">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>635</width>
              <height>574</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item>
              <widget class="QLabel" name="songLabel">
               <property name="text">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:xx-large; font-weight:600;&quot;&gt;No Song&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="filesGroup">
               <property name="title">
                <string>Files</string>
               </property>
               <property name="flat">
                <bool>true</bool>
               </property>
               <layout class="QFormLayout" name="formLayout_2">
                <property name="fieldGrowthPolicy">
                 <enum>QFormLayout::ExpandingFieldsGrow</enum>
                </property>
                <property name="topMargin">
                 <number>9</number>
                </property>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>MP3</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
                  <property name="spacing">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QLineEdit" name="mp3file">
                    <property name="readOnly">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QToolButton" name="setMP3File">
                    <property name="text">
                     <string>...</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_8">
                  <property name="text">
                   <string>Cover</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <layout class="QHBoxLayout" name="horizontalLayout_4">
                  <property name="spacing">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QLineEdit" name="covFile">
                    <property name="readOnly">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QToolButton" name="setCovFile">
                    <property name="text">
                     <string>...</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_10">
                  <property name="text">
                   <string>Video</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="label_9">
                  <property name="text">
                   <string>Background</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <layout class="QHBoxLayout" name="horizontalLayout_5">
                  <property name="spacing">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QLineEdit" name="vidFile">
                    <property name="readOnly">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QToolButton" name="setVidFile">
                    <property name="text">
                     <string>...</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="3" column="1">
                 <layout class="QHBoxLayout" name="horizontalLayout_6">
                  <property name="spacing">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QLineEdit" name="bgFile">
                    <property name="readOnly">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QToolButton" name="setBGFile">
                    <property name="text">
                     <string>...</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="generalGroup">
               <property name="title">
                <string>General</string>
               </property>
               <property name="flat">
                <bool>true</bool>
               </property>
               <layout class="QFormLayout" name="formLayout">
                <property name="fieldGrowthPolicy">
                 <enum>QFormLayout::ExpandingFieldsGrow</enum>
                </property>
                <item row="0" column="0">
                 <widget class="QLabel" name="titleLabel">
                  <property name="text">
                   <string>Title</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLineEdit" name="title"/>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="artistLabel">
                  <property name="text">
                   <string>Artist</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="artist"/>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="editionLabel">
                  <property name="text">
                   <string>Edition</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QLineEdit" name="edition"/>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="genreLabel">
                  <property name="text">
                   <string>Genre</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QLineEdit" name="genre"/>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Messages</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5" stretch="0,0,0,0,0">
         <item>
          <widget class="QScrollArea" name="actionItemsScrollArea">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QAbstractScrollArea::AdjustIgnored</enum>
           </property>
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="actionItems">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>635</width>
              <height>294</height>
             </rect>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="songMessages">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QAbstractScrollArea::AdjustToContents</enum>
           </property>
           <property name="resizeMode">
            <enum>QListView::Adjust</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="redoList">
           <property name="sizeAdjustPolicy">
            <enum>QAbstractScrollArea::AdjustToContents</enum>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::NoSelection</enum>
           </property>
           <property name="resizeMode">
            <enum>QListView::Adjust</enum>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="redoUndo">
           <item>
            <widget class="QPushButton" name="undoButton">
             <property name="text">
              <string>Undo</string>
             </property>
             <property name="icon">
              <iconset theme="go-down">
               <normaloff>.</normaloff>.</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="redoButton">
             <property name="text">
              <string>Redo</string>
             </property>
             <property name="icon">
              <iconset theme="go-up">
               <normaloff>.</normaloff>.</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QListWidget" name="undoList">
           <property name="sizeAdjustPolicy">
            <enum>QAbstractScrollArea::AdjustToContents</enum>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::NoSelection</enum>
           </property>
           <property name="resizeMode">
            <enum>QListView::Adjust</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="mediaTab">
        <attribute name="title">
         <string>Media</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_7" stretch="0,1">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QPushButton" name="popoutVideo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="focusPolicy">
              <enum>Qt::WheelFocus</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../ultraedit.qrc">
               <normaloff>:/images/popout.png</normaloff>
               <normalon>:/images/popin.png</normalon>:/images/popout.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>16</width>
               <height>16</height>
              </size>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Resolution:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="videoResolution">
             <property name="text">
              <string>unknown</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Codec:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="videoCodec">
             <property name="text">
              <string>unknown</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="VideoWidget" name="videoWidget" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tonesTab">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);</string>
        </property>
        <attribute name="title">
         <string>Tones</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QCheckBox" name="checkBox_4">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Play Metronome</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="NoteWidget" name="notes" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="lyricsTab">
        <attribute name="title">
         <string>Lyrics</string>
        </attribute>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QTextEdit" name="rawLyrics"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="rawDataWidget">
        <attribute name="title">
         <string>Raw Data</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QTextEdit" name="rawData">
           <property name="styleSheet">
            <string notr="true">background-color: palette(mid)</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="acceptRichText">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NoteWidget</class>
   <extends>QWidget</extends>
   <header>notewidget.h</header>
   <container>1</container>
   <slots>
    <signal>play()</signal>
    <signal>pause()</signal>
    <signal>seek(quint64)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>VideoWidget</class>
   <extends>QWidget</extends>
   <header>videowidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../ultraedit.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>undoButton_2</sender>
   <signal>clicked()</signal>
   <receiver>undoButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>537</x>
     <y>33</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoButton_2</sender>
   <signal>clicked()</signal>
   <receiver>redoButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>589</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>449</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
